src/wallet/mms.h

```diff
@@ -0,0 +1,292 @@
+    uint32_t member_index_by_monero_address(const cryptonote::account_public_address &monero_address) const;
+    uint32_t member_index_by_transport_address(const std::string &transport_address) const;
+    const std::vector<coalition_member> &get_all_members() const { return m_members; };
+    
+    // Process data just created by "me" i.e. the own local wallet, e.g. as the result of a "prepare_multisig" command
+    // Creates the resulting messages to the right members
+    void process_wallet_created_data(message_type type, const std::string &content);
+    
+    // Go through all the messages, look at the "ready to process" ones, and check whether any single one
+    // or any group of them can be processed, because they are processable as single messages (like a tx
+    // that is fully signed and thus ready for submit to the net) or because they form a complete group
+    // (e.g. key sets from all coalition members to make the wallet multisig). If there are multiple
+    // candidates, e.g. in 2/3 multisig sending to one OR the other member to sign, there will be more
+    // than 1 element in 'data' for the user to choose. If nothing is ready "false" is returned.
+    // The method mostly ignores the order in which the messages were received because messages may be delayed
+    // (e.g. sync data from a member arrives AFTER a transaction to submit) or because message time stamps
+    // may be wrong so it's not possible to order them reliably.
+    // Messages also may be ready by themselves but the wallet not yet ready for them (e.g. sync data already
+    // arriving when the wallet is not yet multisig because key sets were delayed or were lost altogether.)
+    // If nothing is ready 'wait_reason' may contain further info about the reason why.
+    bool get_processable_messages(bool force_sync,
+                                  std::vector<processing_data> &data_list,
+                                  std::string &wait_reason);
+    void set_messages_processed(const processing_data &data);
+    
+    uint32_t add_message(uint32_t member_index, message_type type, message_direction direction, 
+                         const std::string &content);
+    const std::vector<message> &get_all_messages() const { return m_messages; };
+    message get_message_by_id(uint32_t id);
+    void set_message_processed_or_sent(uint32_t id);
+    void delete_message(uint32_t id);
+    void delete_all_messages();
+    
+    // Debugging support: Read and write the contents of single messages into files for "sending" and "receiving" them
+    void send_message(uint32_t id);
+    bool receive_message();
+    
+    void write_to_file(const std::string &filename);
+    void read_from_file(const std::string &filename);
+    
+    template <class t_archive>
+    inline void serialize(t_archive &a, const unsigned int ver)
+    {
+      a & m_active;
+      a & m_coalition_size;
+      if (ver > 0)
```

Comment:
You spotted that with very sharp eyes :)

That's for me during development, so that I do not loose my prepared wallets with filled message stores every time I add some member to those classes. I will reset all versions to 0 before making the "real" PR.